include/boost/geometry/strategies/cartesian/line_interpolate_point.hpp

```diff
@@ -0,0 +1,133 @@
+// Boost.Geometry
+
+// Copyright (c) 2018, Oracle and/or its affiliates.
+
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Licensed under the Boost Software License version 1.0.
+// http://www.boost.org/users/license.html
+
+#ifndef BOOST_GEOMETRY_STRATEGIES_CARTESIAN_LINE_INTERPOLATE_POINT_HPP
+#define BOOST_GEOMETRY_STRATEGIES_CARTESIAN_LINE_INTERPOLATE_POINT_HPP
+
+#include <boost/geometry/core/assert.hpp>
+#include <boost/geometry/core/coordinate_dimension.hpp>
+#include <boost/geometry/core/coordinate_type.hpp>
+#include <boost/geometry/strategies/line_interpolate_point.hpp>
+#include <boost/geometry/strategies/cartesian/distance_pythagoras.hpp>
+#include <boost/geometry/util/select_most_precise.hpp>
+
+
+namespace boost { namespace geometry
+{
+
+namespace strategy { namespace line_interpolate_point
+{
+
+
+/*!
+\brief Interpolate point on a cartesian segment.
+\ingroup strategies
+\tparam CalculationType \tparam_calculation
+\tparam DistanceStrategy The underlying point-point distance strategy
+
+\qbk{
+[heading See also]
+\* [link geometry.reference.algorithms.line_interpolate_point.line_interpolate_point_4_with_strategy line_interpolate_point (with strategy)]
+}
+
+*/
+template
+<
+    typename CalculationType = void,
+    typename DistanceStrategy = distance::pythagoras<CalculationType>
+>
+class cartesian
+{
+public:
+
+    //result type
+    template <typename Point>
+    struct result_type
+    {
+        typedef typename select_most_precise
+            <
+                typename coordinate_type<Point>::type,
+                CalculationType
+            >::type calc_t;
+
+        result_type() :
+            distance(0)
+        {}
+
+        result_type(calc_t d) :
+            distance(d)
+        {}
+
+        calc_t distance;
+    };
+
+    template <typename Point>
+    inline result_type<Point> compute(Point const& p0,
+                                      Point const& p1) const
+    {
+        return result_type<Point>(DistanceStrategy().apply(p0,p1));
+    }
+
+    template <typename Point, typename T1, typename T2>
+    inline void apply(Point const& p0,
+                      Point const& p1,
+                      T1 const& fraction,
+                      Point & p,
+                      T2 const&) const
+    {
+        typedef typename select_most_precise
+            <
+                typename coordinate_type<Point>::type,
+                CalculationType
+            >::type calc_t;
```

Comment:
To be consistent: `select_calculation_type_alt<CalculationType, Point>::type`.

Reply:
Actually you should probably do the same as in other strategies, so: `result_type<Point>::calc_t`.